tests/components/yale/test_binary_sensor.py

```diff
@@ -0,0 +1,390 @@
+                    ),
+                    "url": "https://dyu7azbnaoi74.cloudfront.net/images/1f8.jpeg",
+                    "etag": "09e839331c4ea59eef28081f2caa0e90",
+                },
+                "doorbellName": "Front Door",
+                "callID": None,
+                "origin": "mars-api",
+                "mutableContent": True,
+            },
+        },
+    )
+
+    await hass.async_block_till_done()
+
+    binary_sensor_k98gidt45gul_name_motion = hass.states.get(
+        "binary_sensor.k98gidt45gul_name_motion"
+    )
+    assert binary_sensor_k98gidt45gul_name_motion.state == STATE_ON
+
+    binary_sensor_k98gidt45gul_name_ding = hass.states.get(
+        "binary_sensor.k98gidt45gul_name_doorbell_ding"
+    )
+    assert binary_sensor_k98gidt45gul_name_ding.state == STATE_OFF
+
+    new_time = dt_util.utcnow() + datetime.timedelta(seconds=40)
+    native_time = datetime.datetime.now() + datetime.timedelta(seconds=40)
+    with patch(
+        "homeassistant.components.yale.util._native_datetime",
+        return_value=native_time,
+    ):
+        async_fire_time_changed(hass, new_time)
+        await hass.async_block_till_done()
+
+    binary_sensor_k98gidt45gul_name_image_capture = hass.states.get(
+        "binary_sensor.k98gidt45gul_name_image_capture"
+    )
+    assert binary_sensor_k98gidt45gul_name_image_capture.state == STATE_OFF
+
+    listener(
+        doorbell_one.device_id,
+        dt_util.utcnow(),
+        {
+            "status": "buttonpush",
+        },
+    )
+
+    await hass.async_block_till_done()
+
+    binary_sensor_k98gidt45gul_name_ding = hass.states.get(
+        "binary_sensor.k98gidt45gul_name_doorbell_ding"
+    )
+    assert binary_sensor_k98gidt45gul_name_ding.state == STATE_ON
+    new_time = dt_util.utcnow() + datetime.timedelta(seconds=40)
+    native_time = datetime.datetime.now() + datetime.timedelta(seconds=40)
+    with patch(
+        "homeassistant.components.yale.util._native_datetime",
+        return_value=native_time,
+    ):
+        async_fire_time_changed(hass, new_time)
+        await hass.async_block_till_done()
+
+    binary_sensor_k98gidt45gul_name_ding = hass.states.get(
+        "binary_sensor.k98gidt45gul_name_doorbell_ding"
+    )
+    assert binary_sensor_k98gidt45gul_name_ding.state == STATE_OFF
+
+
+async def test_doorbell_device_registry(
+    hass: HomeAssistant, device_registry: dr.DeviceRegistry
+) -> None:
+    """Test creation of a lock with doorsense and bridge ands up in the registry."""
+    doorbell_one = await _mock_doorbell_from_fixture(hass, "get_doorbell.offline.json")
+    await _create_yale_with_devices(hass, [doorbell_one])
+
+    reg_device = device_registry.async_get_device(identifiers={("yale", "tmt100")})
+    assert reg_device.model == "hydra1"
+    assert reg_device.name == "tmt100 Name"
+    assert reg_device.manufacturer == "Yale Home Inc."
+    assert reg_device.sw_version == "3.1.0-HYDRC75+201909251139"
```

Comment:
Snapshots?

Reply:
switch to snapshots for this one

- [x] yale ce6c7f7b1dd
- [ ] august